src/main/java/net/coderbot/iris/gl/shader/StandardMacros.java

```
@@ -0,0 +1,184 @@
+		String version = SharedConstants.getGameVersion().getReleaseTarget(); //release target so snapshots are set to the higher version
+
+		Matcher matcher = SEMVER_PATTERN.matcher(version);
+
+		if (!matcher.matches()) {
+			return null;
+		}
+
+		String major = group(matcher, "major");
+		String minor = group(matcher, "minor");
+		String bugfix = group(matcher, "bugfix");
+
+		if (bugfix == null) {
+			bugfix = "00";
+		}
+
+		if (minor == null || major == null) return null;
+
+		if (minor.length() == 1) {
+			minor = 0 + minor;
+		}
+		if (bugfix.length() == 1) {
+			bugfix = 0 + bugfix;
+		}
+
+		return major + minor + bugfix;
+	}
+
+	/**
+	 * Returns the current OS String
+	 *
+	 * @return the string based on the current OS
+	 * @see <a href="https://github.com/sp614x/optifine/blob/master/OptiFineDoc/doc/shaders.txt#L687-L692">Optifine Doc</a>
+	 */
+	public static String getOsString() {
+		switch (Util.getOperatingSystem()) {
+			case OSX:
+				return "MC_OS_MAC";
+			case LINUX:
+				return "MC_OS_LINUX";
+			case WINDOWS:
+				return "MC_OS_WINDOWS";
+			case SOLARIS: //Note: Optifine doesn't have a macro for Solaris. https://github.com/sp614x/optifine/blob/master/OptiFineDoc/doc/shaders.txt#L689-L692
+			case UNKNOWN:
+			default:
+				return "MC_OS_UNKNOWN";
+		}
+	}
+
+	/**
+	 * Returns the current GL Version using regex
+	 *
+	 * @param name the name of the gl attribute to parse
+	 * @return current gl version stripped of semantic versioning
+	 * @see <a href="https://github.com/sp614x/optifine/blob/master/OptiFineDoc/doc/shaders.txt#L679-L681">Optifine Doc for GL Version</a>
+	 * @see <a href="https://github.com/sp614x/optifine/blob/master/OptiFineDoc/doc/shaders.txt#L683-L685">Optifine Doc for GLSL Version</a>
+	 */
+	public static String getGlVersion(int name) {
+		String info = GL20.glGetString(name);
+
+		Matcher matcher = SEMVER_PATTERN.matcher(Objects.requireNonNull(info));
+
+		if (!matcher.matches()) {
+			return null;
+		}
```

Similar comments here regarding the regex and silent failure